docs/madr/decisions/076-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,262 @@
+
+### As a Kuma developer I want to rename resources that are coming from outside Kuma
+
+This is either not possible or very hard.
+Envoy can connect to many xDS servers to get configuration from.
+If we integrate with systems like Spire (that exposes its own SDS server) we will not be able to rename resources that are coming from there.
+
+See:
+- https://github.com/spiffe/spire/blob/f687bf21e812a4bf027d88136032fc46497e7fe0/pkg/agent/endpoints/sdsv3/handler.go#L395
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L282
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L291
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L312
+
+## Design
+
+### Use a `^system_([a-z0-9]+_{0,1})+$` regex to name system resources
+
+All changes will be behind the same feature flag as in [Migrating to consistent and well-defined naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
+
+All system resources will conform to `^system_([a-z0-9]+_{0,1})+$` regex (we shouldn't use `_system:` because of [this issue](https://github.com/kumahq/kuma/issues/2363)).
+
+System resources that can be traced back to a Kuma resource with a valid KRI will have take the form of `system_<KRI>`.
+
+For example:
+
+```
+system_kri_mtr_mesh-1__kuma-system_my-meshtrace_
+```
+
+Which indicates a system resource that was created for a `MeshTrace` resource.
+
+Resources that are not correlated with any Kuma resource like:
+
+```
+kuma:envoy:admin
+```
+
+Will conform with the previously mentioned regex:
+
+```
+system_envoy_admin
+```
+
+and should describe the resource as accurately and plainly as possible (it **MUST** take into account any related configuration option or annotation if exists).
+
+It means that `listener.0.0.0.0_15001` which can be configured by (`kuma.io/transparent-proxying-outbound-port`) will become `system_transparent_proxy_outbound_listener`.
+
+#### Secrets
+
+In the future we will most likely introduce new policies - `MeshTrust`, `MeshIdentity` and `SPIRE` integration.
+
+For `Spire` we can't control the secrets as mentioned in [Non use cases section](#as-a-kuma-developer-i-want-to-rename-resources-that-are-coming-from-outside-kuma).
+
+For a secret related to `MeshTrust`s we can't use `KRI` because it's a combination of all of them and `KRI` currently don't deal with lists of resources.
+
+For `MeshIdentity` and user defined secrets we can use `KRI` but that will be covered by [Migrating to consistent and well-defined naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
+
+For all other secrets we can use `^system_([a-z0-9]+_{0,1})+$` naming.
```

Comment:
Isn't all this going in so much details that it confuses the reader.

It seems the rule here is:

> Secrets are resources like any others but should most likely always be considered system. Therefore, they will follow if they are an actual Kuma secret `^system_KRI` and `^system_<someusefulName>` otherwise.
There are possible exceptions but these should be explicitly called out in MADRs. (.e.g [MeshTrust and MeshIdentity modify link](linkToSpiffe)).